sklearn/metrics/cluster/unsupervised.py

```diff
@@ -154,66 +171,38 @@ def silhouette_samples(X, labels, metric='euclidean', **kwds):
        <http://en.wikipedia.org/wiki/Silhouette_(clustering)>`_
 
     """
+    check_array(labels)
     distances = pairwise_distances(X, metric=metric, **kwds)
+    if sample_weight is not None:
```

Comment:
This can also be viewed like,

for a given sample, if another sample belonging to the same cluster has a very high sample weight and is far away, it should reduce the silhoutte score more
right?
And vice versa.